### PR TITLE
Integration of Applicant Class

### PR DESCRIPTION
**Pull request by RyanCheungJF**
Latest update of refactoring codebase from AB3 -> Applicants in LinkedOUT.

As of the latest push by zhikai, all test cases pass! Thanks to those that helped with debugging the test cases.

Some methods/ comments may still have 'AddressBook' or 'Person', do change them if you encounter them!

Updated 9/3 3PM

### CODE REVIEW

File: src/main/java/seedu/linkedout/MainApp.java [hunk 1]
```diff
@@ -74,20 +74,20 @@ public void init() throws Exception {
      * or an empty linkedout book will be used instead if errors occur when reading {@code storage}'s linkedout book.
```
Comment (ryantanlien) [inline src/main/java/seedu/linkedout/MainApp.java:74]: Line 74: linkedout book comment does not make sense here, maybe change the comment wording.

File: src/main/java/seedu/linkedout/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -8,21 +8,21 @@
 import seedu.linkedout.commons.core.index.Index;
 import seedu.linkedout.logic.commands.exceptions.CommandException;
 import seedu.linkedout.model.Model;
-import seedu.linkedout.model.person.Person;
+import seedu.linkedout.model.applicant.Applicant;
 
 /**
- * Deletes a person identified using it's displayed index from the linkedout book.
```
Comment (ryantanlien) [inline src/main/java/seedu/linkedout/logic/commands/DeleteCommand.java:14]: Line 14:
Deletes a applicant -> Deletes an applicant
Linkedout book -> Linkedout

Can consider reviewing the comments on this line.

File: src/main/java/seedu/linkedout/model/Model.java [hunk 3]
```diff
@@ -37,52 +37,52 @@
     /**
      * Returns the user prefs' linkedout book file path.
```
Comment (ryantanlien) [inline src/main/java/seedu/linkedout/model/Model.java:38]: Line 38, 34, 48, 52, 56, 62, 68, 76
Linkedout book -> Linkedout

File: src/main/java/seedu/linkedout/model/UserPrefs.java [hunk 4]
```diff
@@ -51,11 +52,21 @@ public Path getAddressBookFilePath() {
         return addressBookFilePath;
     }
 
+    public Path getLinkedoutFilePath() {
+        // return linkedoutBookFilePath;
+        return addressBookFilePath;
```
Comment (ryantanlien) [inline src/main/java/seedu/linkedout/model/UserPrefs.java:57]: The method getLinkedoutFilePath() still uses the addressBookFilePath, maybe can update to linkedoutBookFilePath?
Not sure if the implementation will still work though. 
If you have time maybe can update this?

File: src/main/java/seedu/linkedout/model/util/SampleDataUtil.java [hunk 5]
```diff
@@ -4,46 +4,46 @@
...
+import seedu.linkedout.model.ReadOnlyLinkedout;
+import seedu.linkedout.model.applicant.Address;
+import seedu.linkedout.model.applicant.Applicant;
+import seedu.linkedout.model.applicant.Email;
+import seedu.linkedout.model.applicant.Name;
+import seedu.linkedout.model.applicant.Phone;
 import seedu.linkedout.model.tag.Tag;
 
 /**
  * Contains utility methods for populating {@code AddressBook} with sample data.
```
Comment (ryantanlien) [inline src/main/java/seedu/linkedout/model/util/SampleDataUtil.java:17]: {@code AddressBook should no longer exist}

This comment contains a direct reference to a file which doesn't exist. Quite a high priority to fix comment-wise.

**Review by ryantanlien — COMMENTED** [review]
Generally a good refactor! There are some minor grammar issues in the comments and JavaDocs regarding linkedout book. 

I don't see any updates to the tests. Do those work? If you need help with rewriting the tests I can help you out as well!

File: src/main/java/seedu/linkedout/model/UserPrefs.java (hunk 4 above)
Comment (RyanCheungJF) [reply src/main/java/seedu/linkedout/model/UserPrefs.java:57]: Hi Ryan, I think its okay to comment it out for now, as linkedoutBookFilePath would point to a different file path and cause the program to crash. I'll update it when storage is done refactoring!

File: src/main/java/seedu/linkedout/logic/commands/DeleteCommand.java (hunk 2 above)
Comment (RyanCheungJF) [reply src/main/java/seedu/linkedout/logic/commands/DeleteCommand.java:14]: Thanks for the spot! Updated.

**Review by zzhikai — COMMENTED** [review]
Looks good so far, guess only left storage refactoring inclusive of tests, and to change to linkedoutfilepath.

**Review by zzhikai — APPROVED** [review]
LGTM! Great work with refactoring and integration so far😊

**Review by ryantanlien — APPROVED** [review]
LGTM! Good work on changing and updating the test cases!

**Pull request merged**